impl/spanner/testutil/testutil.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package testutil helps with tests against Spanner.
+package testutil
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/spannertest"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+var testDBFlag = flag.String("test_db", "", "Fully-qualified database name to test against; empty means use an in-memory fake.")
```

Comment:
Can you describe what happens when this flag is set, and there are multiple unit tests calling `CreateDatabase`? I imagine there is a pre-created DB instance, but I don't see how different unit tests exist together in this database. Should this be a higher level integration test?

Reply:
removed